Source/WebCore/platform/graphics/avfoundation/FormatDescriptionUtilities.cpp

```diff
@@ -245,96 +252,397 @@ bool formatDescriptionIsProtected(CMFormatDescriptionRef formatDescription)
+            return { };
+        CFDataGetBytes(intrinsicMatrix, CFRangeMake(0, sizeof(cameraCalibration.intrinsicMatrix)), reinterpret_cast<UInt8*>(&cameraCalibration.intrinsicMatrix));
+        CFNumberGetValue(intrinsicMatrixProjectionOffset, kCFNumberFloat32Type, &cameraCalibration.intrinsicMatrixProjectionOffset);
+
+        CGSize size = CGSizeZero;
+        if (!CGSizeMakeWithDictionaryRepresentation(intrinsicMatrixReferenceDimensions, &size))
+            return { };
+        cameraCalibration.intrinsicMatrixReferenceDimensions = { size.width, size.height };
+
+        if (auto source = toExtrinsicOriginSource(extrinsicOriginSource))
+            cameraCalibration.extrinsicOriginSource = *source;
+        else
+            return { };
+
+        cameraCalibration.extrinsicOrientationQuaternion = makeVector<float, CFNumberRef>(extrinsicOrientationQuaternion);
+
+        collection.append(WTFMove(cameraCalibration));
+    }
+    return collection;
 }
 
 static std::optional<VideoProjectionMetadataKind> toVideoProjectionMetadataKind(CFStringRef kind)
 {
     if (!kind)
         return { };
 
-    if (PAL::canLoad_CoreMedia_kCMFormatDescriptionProjectionKind_Equirectangular()
-        && CFStringCompare(kind, PAL::kCMFormatDescriptionProjectionKind_Equirectangular, 0) == kCFCompareEqualTo)
-        return VideoProjectionMetadata::Kind::Equirectangular;
+    if (CFStringCompare(kind, PAL::kCMFormatDescriptionProjectionKind_Rectilinear, 0) == kCFCompareEqualTo)
+        return ImmersiveVideoMetadata::Kind::Rectilinear;
 
-    if (PAL::canLoad_CoreMedia_kCMFormatDescriptionProjectionKind_HalfEquirectangular()
-        && CFStringCompare(kind, PAL::kCMFormatDescriptionProjectionKind_HalfEquirectangular, 0) == kCFCompareEqualTo)
-        return VideoProjectionMetadata::Kind::HalfEquirectangular;
+    if (CFStringCompare(kind, PAL::kCMFormatDescriptionProjectionKind_Equirectangular, 0) == kCFCompareEqualTo)
+        return ImmersiveVideoMetadata::Kind::Equirectangular;
 
-    if (PAL::canLoad_CoreMedia_kCMFormatDescriptionProjectionKind_ParametricImmersive()
-        && CFStringCompare(kind, PAL::kCMFormatDescriptionProjectionKind_ParametricImmersive, 0) == kCFCompareEqualTo)
-        return VideoProjectionMetadata::Kind::Parametric;
+    if (CFStringCompare(kind, PAL::kCMFormatDescriptionProjectionKind_HalfEquirectangular, 0) == kCFCompareEqualTo)
+        return ImmersiveVideoMetadata::Kind::HalfEquirectangular;
 
-    if (PAL::canLoad_CoreMedia_kCMFormatDescriptionProjectionKind_AppleImmersiveVideo()
-        && CFStringCompare(kind, PAL::kCMFormatDescriptionProjectionKind_AppleImmersiveVideo, 0) == kCFCompareEqualTo)
-        return VideoProjectionMetadata::Kind::AppleImmersiveVideo;
+    if (CFStringCompare(kind, PAL::kCMFormatDescriptionProjectionKind_ParametricImmersive, 0) == kCFCompareEqualTo)
+        return ImmersiveVideoMetadata::Kind::Parametric;
```

Comment:
So kCMFormatDescriptionProjectionKind_ParametricImmersive corresponds ImmersiveVideoMetadata::Kind::Parametric, is the shorter name without "Immersive" intended? (Other ones seem to match fully.)

Reply:
I'm just re-using the existing enum here.